src/aggregator/integration/placement_change_test.go

```diff
@@ -237,45 +222,42 @@ func TestPlacementChange(t *testing.T) {
 		clock.SetNow(data.timestamp)
 
 		for _, mm := range data.metricWithMetadatas {
-			idx := getServerIndex(mm.metric.ID(), initialPlacement)
-			require.NoError(t, clients[idx].writeTimedMetricWithMetadata(mm.metric.timed, mm.metadata.timedMetadata))
-		}
-		for _, c := range clients {
-			require.NoError(t, c.flush())
+			require.NoError(t, client.writeTimedMetricWithMetadata(mm.metric.timed, mm.metadata.timedMetadata))
 		}
+		require.NoError(t, client.flush())
 
 		// Give server some time to process the incoming packets.
 		time.Sleep(sleepDuration)
 	}
 
 	clock.SetNow(start2)
-	time.Sleep(sleepDuration)
-	require.NoError(t, setPlacement(placementKey, kvStore, finalPlacement))
-	time.Sleep(sleepDuration)
+	time.Sleep(6 * time.Second)
+	require.NoError(t, setPlacementWithClusterClient(placementKey, clusterClient, finalPlacement))
+	time.Sleep(6 * time.Second)
```

Comment:
All those 6 second sleeps feel gross and somewhat arbitrary. What is the condition we are waiting for here? 
Maybe it would be possible to use `xclock.WaitUntil` with some appropriate check (eg. placement updated) instead of sleep?

Reply:
First sleep is waiting for flush to happen. Second one – for the placement update to be processed. Not sure if there is a straightforward way to check if these events have happened.